src/create-file/index.js

```diff
@@ -0,0 +1,62 @@
+import fsPromises from 'fs/promises';
+import { initialize, log } from '../workerTemplate.js';
+const workerQueue = process.env.WORKERQUEUE;
+const resultQueue = process.env.RESULTSQUEUE;
+const rabbitHost = process.env.RABBITHOST;
+const rabbitUser = process.env.RABBITUSER;
+const rabbitPass = process.env.RABBITPASS;
+
+/**
+ * Creates a file at the given destination path and with the given content.
+ * @param {Object} workerJob The job object
+ * @param {Array} inputs The inputs for this process
+ *   First input {String|Buffer|TypedArray|DataView} is the content of the file to be written
+ *   Second input {String} is the absolute path to store the content to. If filename and suffix
+ *     are ommited, they will be generated (file.txt). Full path is returned as output argument.
+ */
+const callbackWorker = async (workerJob, inputs) => {
+    const content = inputs[0];
+    const destinationFileName = inputs[1];
+    const hasSuffix = destinationFileName.includes(".") &&
+      destinationFileName.lastIndexOf(".") < destinationFileName.length - 1;
+    let finalDestinationFileName = destinationFileName;
+
+    if (!content) {
+        throw "No content given to create-file worker";
+    }
+
+    if (!finalDestinationFileName) {
+        throw "No destination file path given to create-file worker";
+    }
+
+    if (!hasSuffix) {
+        try {
+            await fsPromises.opendir(finalDestinationFileName);
+            if (destinationFileName.endsWith("/")) {
+                finalDestinationFileName = destinationFileName + "file.txt";
+            } else {
+                finalDestinationFileName = destinationFileName + "/file.txt";
+            }
+        } catch (err) {
+            throw "Could not open folder " + finalDestinationFileName;
+        }
+    }
+
+    log('Creating file ' + finalDestinationFileName + ' …');
+    await fsPromises.writeFile(finalDestinationFileName, content);
+
+    log('File created successfully.');
+    workerJob.status = 'success';
+    workerJob.outputs = [finalDestinationFileName];
+};
+
+(async () => {
+    try {
+        // Initialize and start the worker process
+        await initialize(rabbitHost, rabbitUser, rabbitPass, workerQueue, resultQueue, callbackWorker);
+    } catch (e) {
+        log('Problem when initializing:', e);
```

Comment:
comma does not work in this custom log statement
```suggestion
        log('Problem when initializing:' + e);
```